x-pack/plugins/lens/public/shared_components/legend_settings_popover.tsx

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiFormRow, EuiButtonGroup, EuiSpacer, EuiSwitch, EuiSwitchEvent } from '@elastic/eui';
+import { Position } from '@elastic/charts';
+import { ToolbarPopover } from '../shared_components';
+
+interface LegendSettingsPopoverProps {
+  /**
+   * Determines the legend display options
+   */
+  legendOptions: Array<{ id: string; value: 'auto' | 'show' | 'hide' | 'default'; label: string }>;
+  /**
+   * Determines the legend mode
+   */
+  mode: 'default' | 'show' | 'hide' | 'auto';
+  /**
+   * Callback on display option change
+   */
+  onDisplayChange: (id: string) => void;
+  /**
+   * Sets the legend position
+   */
+  position?: Position;
+  /**
+   * Callback on position option change
+   */
+  onPositionChange: (id: string) => void;
+  /**
+   * If true, nested legend switch is rendered
+   */
+  renderNestedLegendSwitch?: boolean;
+  /**
+   * nested legend switch status
+   */
+  nestedLegend?: boolean;
+  /**
+   * Callback on nested switch status change
+   */
+  onNestedLegendChange?: (event: EuiSwitchEvent) => void;
+}
+
+const toggleButtonsIcons = [
+  {
+    id: Position.Bottom,
+    label: 'Bottom',
+    iconType: 'arrowDown',
+  },
+  {
+    id: Position.Left,
+    label: 'Left',
+    iconType: 'arrowLeft',
+  },
+  {
+    id: Position.Right,
+    label: 'Right',
+    iconType: 'arrowRight',
+  },
+  {
+    id: Position.Top,
+    label: 'Top',
+    iconType: 'arrowUp',
+  },
+];
+
+export const LegendSettingsPopover: React.FunctionComponent<LegendSettingsPopoverProps> = ({
+  legendOptions,
+  mode,
+  onDisplayChange,
+  position,
+  onPositionChange,
+  renderNestedLegendSwitch = false,
+  nestedLegend = false,
+  onNestedLegendChange = () => {},
+}) => {
+  return (
+    <ToolbarPopover
+      title={i18n.translate('xpack.lens.xyChart.legendLabel', {
+        defaultMessage: 'Legend',
+      })}
+      type="legend"
+      groupPosition="right"
+    >
+      <EuiFormRow
+        display="columnCompressed"
+        label={i18n.translate('xpack.lens.xyChart.legendVisibilityLabel', {
+          defaultMessage: 'Display',
+        })}
+      >
+        <>
+          <EuiButtonGroup
+            isFullWidth
+            legend={i18n.translate('xpack.lens.xyChart.legendVisibilityLabel', {
+              defaultMessage: 'Display',
+            })}
+            name="legendDisplay"
+            buttonSize="compressed"
+            options={legendOptions}
+            idSelected={legendOptions.find(({ value }) => value === mode)!.id}
+            onChange={onDisplayChange}
+          />
+          <EuiSpacer size="s" />
+          {renderNestedLegendSwitch && (
+            <EuiSwitch
```

Comment:
This EuiSwitch is part of the same EuiFormRow as the other buttons. I think it should be part of its own EuiFormRow.

Reply:
I haven't changed the layout, I kept it as it was on piechart. If we move it to its own row it will change the layout as:
<img width="334" alt="Screenshot 2020-09-09 at 10 01 00 AM" src="https://user-images.githubusercontent.com/17003240/92565535-76715d80-f283-11ea-9956-7837d21e794a.png">

Are you ok to change it like this? cc @cchaos